### ISSUE TITLE
Add support for shadow size (like in CSS3)

### ISSUE THREAD

**Issue by spyridonas** [kangax/fabric.js#1214]
Is it possible to make the shadow larger instead of more blurred? Can you mark it as a feature if not possible ? Like how the box-shadow css 3 property works

**Comment by asturur** [comment]
we do not have much control of shadow on canvas. we cannot make it spread from 2 size nor make it bigger. unless we implement a full shadow engine wit boxes and blur.

**Comment by Saberjsd** [comment]
@asturur Hello, I want use shadow spread in canvas too.
But it doesn't seem to support it in canvas 2D api.
I tried use a sacle larger shadow,but the performance was not well in sharp corners.
Is there a new solution to achieve it?
Thanks every much!

**Comment by asturur** [comment]
no we are not looking in to this either.
For now we can just use the same kind of shadow that is allowed from canvas.
Can opacity help you achieving the same effect?

**Comment by pranavgawri** [comment]
much needed! It will bring a new dimension to the UI generated on Canvas. Please look into it.

**Comment by melchiar** [comment]
@pranavgawri we're limited by the draw functions available in canvas2d, and currently canvas shadows don't have a spread option. 

while you could probably achieve a spread effect it for simple objects like rectangles just by drawing a larger rectangle for the shadow, but I don't know of any way you could do it for complex objects like text. Do you?

**Comment by yafkari** [comment]
Indeed, it is very frustrating to not being able to make the shadow bigger. Let's say I set a black shadow, my only way is to increase the blur but this will "diffuse" the black.

If for example I wanted to get closer to that result:
![image](https://github.com/fabricjs/fabric.js/assets/41365655/763eebd5-f5c4-4b04-88bd-63439fd13846)

I could only reach that result with the options in fabric.Shadow at the moment:
![image](https://github.com/fabricjs/fabric.js/assets/41365655/e331cc46-4d4a-424d-8ae8-5bfe63f5962c)

See how the black is diffused. And obviously if i decrease the blur, it's not that better

![image](https://github.com/fabricjs/fabric.js/assets/41365655/b2b7aca5-226c-45d8-b330-6c3ad26aee9b)

I guess a could add a smaller copy of the text behind, but not a big fan of that workaround in this case.
